freeipa/src/test/java/com/sequenceiq/freeipa/service/freeipa/flow/FreeIpaPostInstallServiceTest.java

```diff
@@ -70,11 +101,122 @@ public void postInstallFreeIpaExecutePostServiceDeploymentRecipes() throws Excep
+        when(ipaClient.userFind("adminka")).thenReturn(Optional.of(user));
+
+        underTest.postInstallFreeIpa(1L, true);
+
+        verify(hostOrchestrator, never()).postServiceDeploymentRecipes(eq(gatewayConfig), eq(nodes), any(StackBasedExitCriteriaModel.class));
+        verify(userSyncBindUserService).createUserAndLdapConfig(stack);
+        verify(wiamClient).syncUsersInEnvironment(eq(ACCOUNT_ID), eq(ENVIRONMENT_CRN), anyString());
+        verifyNoInteractions(userSyncService);
+        verify(ipaClient).addPasswordExpirationPermission("Set Password Expiration");
+        verify(ipaClient).addPermissionToPrivilege("User Administrators", "Set Password Expiration");
+        verify(freeIpaPermissionService).setPermissions(stack, ipaClient);
+        verify(ipaClient).setUsernameLength(255);
+        verify(passwordPolicyService).updatePasswordPolicy(ipaClient);
+        verify(ipaClient).updateUserPasswordMaxExpiration(user.getUid());
+    }
+
+    @Test
+    public void testInternalUsersyncInvoked() throws Exception {
+        Stack stack = mock(Stack.class);
+        when(stack.getAccountId()).thenReturn(ACCOUNT_ID);
+        when(stack.getEnvironmentCrn()).thenReturn(ENVIRONMENT_CRN);
+        GatewayConfig gatewayConfig = mock(GatewayConfig.class);
+        when(stackService.getByIdWithListsInTransaction(1L)).thenReturn(stack);
+        Set<Node> nodes = Set.of(mock(Node.class));
+        when(freeIpaRecipeService.hasRecipeType(1L, RecipeType.POST_SERVICE_DEPLOYMENT, RecipeType.POST_CLUSTER_INSTALL)).thenReturn(false);
+        when(entitlementService.isWorkloadIamSyncEnabled(ACCOUNT_ID)).thenReturn(false);
+        FreeIpaClient ipaClient = mock(FreeIpaClient.class);
+        when(freeIpaClientFactory.getFreeIpaClientForStack(stack)).thenReturn(ipaClient);
+        when(ipaClient.findPermission("Set Password Expiration")).thenReturn(Set.of());
+        when(ipaClient.getUsernameLength()).thenReturn(5);
+        when(freeIpaClientFactory.getAdminUser()).thenReturn("adminka");
+        User user = new User();
+        user.setUid("adminid");
+        when(ipaClient.userFind("adminka")).thenReturn(Optional.of(user));
+
+        ThreadBasedUserCrnProvider.doAs(USER_CRN, () -> {
+            try {
+                underTest.postInstallFreeIpa(1L, true);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        verify(hostOrchestrator, never()).postServiceDeploymentRecipes(eq(gatewayConfig), eq(nodes), any(StackBasedExitCriteriaModel.class));
+        verify(userSyncBindUserService).createUserAndLdapConfig(stack);
+        verify(userSyncService).synchronizeUsers(eq(ACCOUNT_ID), anyString(), eq(Set.of(ENVIRONMENT_CRN)), eq(Set.of()), eq(Set.of()),
+                eq(WorkloadCredentialsUpdateType.UPDATE_IF_CHANGED));
+        verifyNoInteractions(wiamClient);
+        verify(ipaClient).addPasswordExpirationPermission("Set Password Expiration");
```

Comment:
what do you think of factoring the mocking and verifications  for permissions since they are repeated each test?
In fact, these verifications aren't really relevant to these test cases, so they make the test cases harder to read and understand.